iocore/net/SSLNetVConnection.cc

```diff
@@ -322,76 +322,69 @@ ssl_read_from_net(SSLNetVConnection *sslvc, EThread *lthread, int64_t &ret)
 int64_t
 SSLNetVConnection::read_raw_data()
 {
-  int64_t r      = 0;
-  int64_t toread = INT_MAX;
-
   // read data
-  int64_t rattempted = 0, total_read = 0;
-  unsigned niov = 0;
+  int64_t r          = 0;
+  int64_t total_read = 0;
+  int64_t rattempted = 0;
+  int niov           = 0;
   IOVec tiovec[NET_MAX_IOV];
-  if (toread) {
-    IOBufferBlock *b = this->handShakeBuffer->first_write_block();
-    do {
-      niov       = 0;
-      rattempted = 0;
-      while (b && niov < NET_MAX_IOV) {
-        int64_t a = b->write_avail();
-        if (a > 0) {
-          tiovec[niov].iov_base = b->_end;
-          int64_t togo          = toread - total_read - rattempted;
-          if (a > togo)
-            a                  = togo;
-          tiovec[niov].iov_len = a;
-          rattempted += a;
-          niov++;
-          if (a >= togo)
-            break;
-        }
-        b = b->next.get();
-      }
-
-      // If there was no room to write into the buffer then skip the read
-      if (niov > 0) {
-        ink_assert(niov > 0);
-        ink_assert(niov <= countof(tiovec));
-        r = socketManager.readv(this->con.fd, &tiovec[0], niov);
+  IOBufferBlock *b = this->handShakeBuffer->first_write_block();
 
-        NET_INCREMENT_DYN_STAT(net_calls_to_read_stat);
-        total_read += rattempted;
-      } else { // No more space to write, break out
-        r = 0;
-        break;
+  do {
+    niov       = 0;
+    rattempted = 0;
+    while (b && niov < NET_MAX_IOV) {
+      int64_t a = b->write_avail();
+      if (a > 0) {
+        tiovec[niov].iov_base = b->_end;
+        tiovec[niov].iov_len  = a;
+        rattempted += a;
+        niov++;
       }
-    } while (rattempted && r == rattempted && total_read < toread);
-
-    // if we have already moved some bytes successfully, summarize in r
-    if (total_read != rattempted) {
-      if (r <= 0)
-        r = total_read - rattempted;
-      else
-        r = total_read - rattempted + r;
+      b = b->next.get();
     }
-    // check for errors
-    if (r <= 0) {
-      if (r == -EAGAIN || r == -ENOTCONN) {
-        NET_INCREMENT_DYN_STAT(net_calls_to_read_nodata_stat);
-      }
-      return r;
+
+    if (niov == 1) {
+      r = socketManager.read(this->con.fd, tiovec[0].iov_base, tiovec[0].iov_len);
+    } else {
+      r = socketManager.readv(this->con.fd, &tiovec[0], niov);
```

Comment:
yes, I hadn't completely unwound the simplification of this method.  Will push it through.